scrapy/utils/gz.py

```diff
@@ -4,30 +4,38 @@
     from cStringIO import StringIO as BytesIO
 except ImportError:
     from io import BytesIO
-
+from io import UnsupportedOperation
 from gzip import GzipFile
 
+class ReadOneGzipFile(GzipFile):
+    def readone(self, size=-1):
+        try:
+            return self.read1(size)
+        except UnsupportedOperation:
+            return self.read(size)
```

Comment:
Thanks for the explanation!
We don't support Python 3.2, does it help?
Also, what do you think about making readone a function, not a class method? That's one less line of code :)